src/components/reviews/Reviews.js

```diff
@@ -0,0 +1,63 @@
+import { Review } from "./Review";
+import { useState, useEffect } from "react";
+import styles from "@/styles/reviews.module.scss";
+import { useContext } from "react";
+import { LoggedInContext } from "@/pages/_app";
+
+export const Reviews = (props) => {
+  const { handleShowLogInModal, isLoggedIn } = useContext(LoggedInContext);
+  const [reviews, setReviews] = useState([]);
+  const [once, setOnce] = useState(false);
+
+  const handleLogin = () => {
+    window.scrollTo(0, 0);
+    handleShowLogInModal();
+  };
+  if (props.posted && once == true) {
+    setOnce(false);
+  }
+  useEffect(() => {
+    if (!once) {
+      const movieId = window.location.pathname.split("/")[2];
+      getData(movieId);
+      props.setPosted(false);
+      setOnce(true);
+    }
+    async function getData(movieId) {
+      const res = await fetch("/api/movies/" + movieId + "/reviews");
+      if (!res.ok) {
+        throw new Error("Failed to fetch data");
+      }
+      const reviews = await res.json();
+      setReviews(
+        reviews.map(
+          (review) =>
+            (review = (
```

Comment:
This assignment does nothing, because you are not using this variable again in this code. It works because assigning somethign also evaluates to the value of the assignment, e.g.: 

```ts
let x = 5;
const y = (x = 6);
// Now both x and y are 6
```

This code could have been written without line 35 (and removing the corresponding parentheses on line 42).

But the thing I'm really concerned about here is the fact that you are storing react elements (returned by JSX) in state. There should be no need to have the `reviews`/`setReviews` state if all you're doing is storing React elements in it and then returning it.

Just generate the JSX using `map` (like above here) when it's time to render towards the end of the component, e.g. on line 61.